text/0036-reserved-subaddresses.md

```diff
@@ -0,0 +1,68 @@
+- Feature Name: Reserved Subaddresses
+- Start Date: (2022-04-10)
+- MCIP PR: [mobilecoinfoundation/mcips#36](https://github.com/mobilecoinfoundation/mcips/pull/36)
+- Tracking Issue: https://github.com/mobilecoinfoundation/mobilecoin/issues/1787
+
+[summary]: #summary
+
+Subaddress indices are unsigned 64-bit numbers.
+
+The indices with high bit set are reserved for "special purposes", and should
+not be assigned to individual contacts by desktop wallets that are using subaddresses this way.
+
+`u64::MAX` will be reserved for an "invalid / none" value.
+
+[mobilecoinfoundation/mcips#4](https://github.com/mobilecoinfoundation/mcips/pull/4) will be updated,
+changing the change subaddress from `1` to `u64::MAX - 1`.
```

Comment:
Why would we change the change subaddress?

Reply:
It seems like a high cost to change, as this has been deployed as a subaddress for change for more than a year

Reply:
Just synced with @garbageslam offline, and here's where full-service is setting it:

https://github.com/mobilecoinofficial/full-service/blob/main/full-service/src/db/account.rs#L25

Reply:
@sugargoat the problem is that, mobilecoind was already using this subaddress for customers, and they were in the wild even before full-service.

the plan right now is that we would like for mobilecoind users to move to full-service. But it's impossible for them to do that if full-service is treating this subaddress as change. 

* They cannot claw back subaddresses that they already gave to users, which users might continue to deposit money into
* If the change subaddress is known by anyone, then that person is able to forge destination memos in the RTH scheme. So it is very unsafe for an exchange to start using subaddress 1 as a change address.

Since it is security critical that the change subaddress is not known by anyone, and some clients were using subaddress 1 casually, the only safe option is to use a completely new subaddress as the change subaddress.

Reply:
we can mention this in the drawbacks / rationale section